java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileWriter.java

```diff
@@ -208,6 +210,10 @@ private void writeBatch(VectorSchemaRoot recordBatch) throws IOException {
   }
 
   private void writeFromVectorIntoJson(Field field, FieldVector vector) throws IOException {
+    if (vector instanceof ExtensionTypeVector) {
```

Comment:
I'm curious why this is needed, because roundtripping works in our integration tests:
https://github.com/apache/arrow/blob/b99b00dd66586cf54b04ce6a51eb1cf68b1510a3/dev/archery/archery/integration/datagen.py#L1845-L1861

Reply:
@pitrou that's strange :thinking: 

The test in this PR fails without the associated code change. It throws `UnsupportedOperation minor type: EXTENSIONTYPE`, from [here](https://github.com/apache/arrow/blob/72d20ad719021c5513620e23a0a65fb724f0e299/java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileWriter.java#L432) - which seems correct, there's no case in there for EXTENSIONTYPE.

Reply:
Ah! It's possible that the integration tests are only every reading JSON files from Java, then. The JSON files are typically generated by the Python machinery I quoted an excerpt from above.